src/Controller/Component/AuthComponent.php

```diff
@@ -877,7 +877,7 @@ public function constructAuthenticate()
      *
      * @param \Cake\Auth\Storage\StorageInterface|null $storage Sets provided
      *   object as storage or if null returns configured storage object.
-     * @return \Cake\Auth\Storage\StorageInterface|null
+     * @return \Cake\Auth\Storage\StorageInterface|\Cake\Core\InstanceConfigTrait|null
```

Comment:
Unfortunately, this is needed for getConfig() and setConfig calls on this storage() to be typehinted and found as valid.